tests/test_decryption_key_index.rs

```diff
@@ -0,0 +1,60 @@
+use openmls::prelude::*;
+
+mod utils;
+
+use utils::managed_utils::*;
+
+#[test]
+fn test_decryption_key_index_computation() {
+    // Some basic setup functions for the managed group.
+    let handshake_message_format = HandshakeMessageFormat::Plaintext;
+    let update_policy = UpdatePolicy::default();
+    let callbacks = default_callbacks::default_callbacks();
+    let managed_group_config =
+        ManagedGroupConfig::new(handshake_message_format, update_policy, callbacks);
+    let number_of_clients = 20;
+    let setup = ManagedTestSetup::new(managed_group_config, number_of_clients);
+    setup.create_clients();
+
+    for ciphersuite in Config::supported_ciphersuites() {
+        // Create a basic group with more than 4 members to create a tree with intermediate nodes.
+        let group_id = setup.create_random_group(10, ciphersuite).unwrap();
+        let mut groups = setup.groups.borrow_mut();
+        let group = groups.get_mut(&group_id).unwrap();
+
+        // Now we have to create a situation, where the resolution is neither
+        // the leaf, nor the common ancestor closest to the root. To do that, we
+        // first have the member at index 0 remove the one at index 2, thus
+        // populating its own parent node.
+        let (_, remover_id) = &group.members[0].clone();
```

Comment:
This looks like it might be nice to be able to remove a client with an index instead of the id. If we need something like this more often, we might want to add this to the test framework.

Reply:
True. I'll add it.

Reply:
Done!